addons/l10n_it_edi/models/account_move.py

```diff
@@ -0,0 +1,1212 @@
+            new_state = state_map.get((sdi_state, outcome), False)
+
+        parsed_notification.update({
+            'l10n_it_edi_state': new_state,
+            'l10n_it_edi_transaction': False if new_state in (False, 'rejected') else self.l10n_it_edi_transaction,
+            'send_ack_to_edi_proxy': bool(new_state),
+            'date': date,
+            'errors': errors,
+            'filename': filename,
+        })
+        return parsed_notification
+
+    def _l10n_it_edi_write_send_state(self, transformed_notification, message):
+        """ Update the record with the data coming from the IAP server.
+            Eventually post the message.
+            Commit the transaction.
+        """
+        self.ensure_one()
+        old_state = self.l10n_it_edi_state
+        new_state = transformed_notification['l10n_it_edi_state']
+        self.write({
+            'l10n_it_edi_state': new_state,
+            'l10n_it_edi_transaction': transformed_notification['l10n_it_edi_transaction'],
+            'l10n_it_edi_header': message or False,
+        })
+
+        if message and old_state != new_state:
+            self.with_context(no_new_invoice=True).sudo().message_post(body=message)
+
+        if new_state == 'rejected':
+            self.l10n_it_edi_attachment_file = False
+
+        self.env.cr.commit()
+
+    def _l10n_it_edi_get_message(self, transformed_notification):
+        """ The status change will be notified in the chatter of the move.
+            Compute the message from the notification information coming from the EDI Proxy Server
+        """
+        self.ensure_one()
+        partner = self.commercial_partner_id
+        partner_name = partner.display_name
+        filename = transformed_notification['filename']
+        new_state = transformed_notification['l10n_it_edi_state']
+        if new_state == 'rejected':
+            DUPLICATE_MOVE = '00404'
+            DUPLICATE_FILENAME = '00002'
+            error_descriptions = []
+            for error_code, error_description in transformed_notification['errors']:
+                error_description_copy = error_description
+                if error_code == DUPLICATE_MOVE:
+                    error_description_copy = _(
+                        "The e-invoice file %s is duplicated.\n"
+                        "Original message from the SdI: %s",
+                        filename, error_description_copy)
+                elif error_code == DUPLICATE_FILENAME:
+                    error_description_copy = _(
+                        "The e-invoice filename %s is duplicated. Please check the FatturaPA Filename sequence.\n"
+                        "Original message from the SdI: %s",
+                        filename, error_description_copy)
+                error_descriptions.append(error_description_copy)
+
+            return self._l10n_it_edi_format_errors(_('The e-invoice has been refused by the SdI.'), error_descriptions)
+
+        elif partner._is_pa():
```

Comment:
Plz rename this method on partner... `_l10n_it_edi_is_...` I don't know what `pa` is supposed to mean

Reply:
renamed to `_l10n_it_edi_is_public_administration()`.